solutions/search/agent-builder/models.md

```diff
@@ -0,0 +1,154 @@
+2. Select **Create Connector** and select your model provider
+3. Configure the connector with your API credentials and preferred model
+4. Search for **GenAI Settings** in the global search field
+5. Select your new connector from the **Default AI Connector** dropdown
+6. Save your changes
+
+For detailed instructions on creating connectors, refer to [Connectors](https://www.elastic.co/docs/deploy-manage/manage-connectors).
+
+Learn more about [preconfigured connectors](https://www.elastic.co/docs/reference/kibana/connectors-kibana/pre-configured-connectors).
+
+## Connectors API
+
+For programmatic access to connector management, refer to the [Connectors API documentation]({{kib-serverless-apis}}group/endpoint-connectors).
+
+## Recommended models
+
+{{agent-builder}} requires models with strong reasoning and tool-calling capabilities. State-of-the-art models perform significantly better than smaller or older models.
+
+### Recommended model families
+
+The following models are known to work well with {{agent-builder}}:
+
+- **OpenAI**: GPT-4.1, GPT-4o
+- **Anthropic**: Claude Sonnet 4.5, Claude Sonnet 4, Claude Sonnet 3.7
+- **Google**: Gemini 2.5 Pro
+
+### Why model quality matters
+
+Agent Builder relies on advanced LLM capabilities including:
+
+- **Function calling**: Models must accurately select appropriate tools and construct valid parameters from natural language requests
+- **Multi-step reasoning**: Agents need to plan, execute, and adapt based on tool results across multiple iterations
+- **Structured output**: Models must produce properly formatted responses that the agent framework can parse
+
+Smaller or less capable models may produce errors like:
+
+```console-response
+Error: Invalid function call syntax
+```
+
+```console-response
+Error executing agent: No tool calls found in the response.
+```
+
+While any chat-completion-compatible connector can technically be configured, we strongly recommend using state-of-the-art models for reliable agent performance.
+
+:::{note}
+GPT-4o-mini and similar smaller models are not recommended for {{agent-builder}} as they lack the necessary capabilities for reliable agent workflows.
+:::
+
+## Connect a local LLM
+
+You can connect a locally hosted LLM to Elastic using the OpenAI connector. This requires your local LLM to be compatible with the OpenAI API format.
+
+### Requirements
+
+**Model selection:**
+- Must include "instruct" in the model name to work with Elastic
+- Download from trusted sources only
+- Consider parameter size, context window, and quantization format for your needs
+
+**Integration setup:**
+- For Elastic Cloud: Requires a reverse proxy (such as Nginx) to authenticate requests using a bearer token and forward them to your local LLM endpoint
+- For self-managed deployments on the same host as your LLM: Can connect directly without a reverse proxy
+- Your local LLM server must use the OpenAI SDK for API compatibility
+
+### Configure the connector
```

Comment:
TBH it feels like we should have platform-level (solution agnostic) documentation how to setup such connectors. I see similar docs for security and o11y solutions